gamemodes/terrortown/gamemode/client/cl_player_ext.lua

```diff
@@ -412,19 +423,71 @@ hook.Add("CalcView", "TTT2ViewBobbingHook", function(ply, origin, angles, fov)
         return
     end
 
-    if (not ply:IsOnGround() and ply:WaterLevel() == 0) or ply:InVehicle() then
-        airtime = math.Clamp(airtime + 1, 0, 300)
-
-        return
-    end
-
     local view = {
         ply = ply,
         origin = origin,
         angles = angles,
         fov = fov,
     }
 
+    ply.lastFOVValue = ply.lastFOVValue or fov
+
+    local dynFOV = fov
+    local mul = ply:GetSpeedMultiplier() * SPRINT:HandleSpeedMultiplierCalculation(ply)
+    local desiredFOV = fov * mul ^ (1 / 6)
+
+    -- fixed mode: when SetZoom is set to something different than 0, this value should be used
+    -- without any custom modification
+    if ply:GetFOVFixedValue() then
+        dynFOV = fov
+
+    -- dynamic mode: transition between different FOV values
+    else
+        local time = math.max(
+            0,
+            (CurTime() - ply:GetFOVTime()) * game.GetTimeScale() * cvHostTimescale:GetFloat()
```

Comment:
This is taken from the ironsights code. GMod has two variables. A convar that can be set by the server owner and an internal API that can be used by addons. The code should respect both.

As this is used for our custom ironsight zoom-out I thought it would be appropriate to support this here as well.